superset/sql_lab.py

```diff
@@ -108,6 +108,15 @@ def handle_error(msg):
         result_proxy = engine.execute(query.executed_sql, schema=query.schema)
     except Exception as e:
         logging.exception(e)
+        if hasattr(e, 'orig') \
+           and type(e.orig).__name__ == 'DatabaseError' \
```

Comment:
Thanks for the fix !
this code should be enginer specific, could u please move it here:
https://github.com/airbnb/superset/blob/master/superset/db_engine_specs.py

You can call it `extract_error_message` or smth like that
Please add a simple unit test too